tests/fixture/tmpnet/node_process.go

```diff
@@ -0,0 +1,258 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package tmpnet
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/ava-labs/avalanchego/api/health"
+	"github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/node"
+)
+
+const (
+	AvalancheGoPathEnvName = "AVALANCHEGO_PATH"
+
+	defaultNodeInitTimeout = 10 * time.Second
+)
+
+var errNodeAlreadyRunning = errors.New("failed to start node: node is already running")
+
+func checkNodeHealth(ctx context.Context, uri string) (bool, error) {
+	// Check that the node is reporting healthy
+	health, err := health.NewClient(uri).Health(ctx, nil)
+	if err == nil {
+		return health.Healthy, nil
+	}
+
+	switch t := err.(type) {
+	case *net.OpError:
+		if t.Op == "read" {
+			// Connection refused - potentially recoverable
+			return false, nil
+		}
+	case syscall.Errno:
+		if t == syscall.ECONNREFUSED {
+			// Connection refused - potentially recoverable
+			return false, nil
+		}
+	}
+	// Assume all other errors are not recoverable
+	return false, fmt.Errorf("failed to query node health: %w", err)
```

Comment:
In general in avalanchego we don't check whether network errors are potentially temporary/recoverable, and just assume they're not. I'd be ok doing the same here